administration/src/bp-modules/self-service/CardSelfServiceActivation.tsx

```diff
@@ -17,15 +20,21 @@ const StyledIconTextButton = styled(IconTextButton)`
 `
 
 type CardSelfServiceActivationProps = {
-  deepLink: string
-  downloadPdf: () => Promise<void>
+  code: CreateCardsResult
+  downloadPdf: (code: CreateCardsResult, fileName: string) => Promise<void>
 }
 
-const CardSelfServiceActivation = ({ deepLink, downloadPdf }: CardSelfServiceActivationProps): ReactElement => {
+const CardSelfServiceActivation = ({ code, downloadPdf }: CardSelfServiceActivationProps): ReactElement => {
+  const projectConfig = useContext(ProjectConfigContext)
   const { t } = useTranslation('selfService')
+  const deepLink = getCustomDeepLinkFromQrCode(projectConfig, {
+    case: 'dynamicActivationCode',
+    value: code.dynamicActivationCode,
+  })
+
   return (
     <Container>
-      <StyledIconTextButton onClick={downloadPdf}>
+      <StyledIconTextButton onClick={() => downloadPdf(code, `${projectConfig.name}.pdf`)}>
```

Comment:
:question: So far the pdf name was only `projectConfig.name`, was that intended?

Reply:
hm if you test on recent staging for koblenz, you download a file that is a working pdf with the correct type extension

Reply:
Hmm, maybe the browser automatically adds the extension based on the file type? Anyway, I think it does not hurt to make it explicit right?

The previous code:
``` ts
await downloadPdf(code, projectConfig.name)
```
with
```
  const downloadPdf = async (code: CreateCardsResult, fileName: string): Promise<void> => {
    const blob = await generatePdf([code], [selfServiceCard], projectConfig.pdf)
    downloadDataUri(blob, fileName)
  }
```

Reply:
looks good to me